pkg/events/events.go

```diff
@@ -0,0 +1,120 @@
+package events
+
+import (
+	k8sV1 "k8s.io/api/core/v1"
+
+	submV1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+)
+
+const AnyNetworkPlugin = "any"
+
+type EventHandler interface {
+
+	// Init initializes the driver with any state it needs.
+	// 	    Called once during startup, to let the plugin configure any necessary settings on the
+	//	    system once the route-agent is ready
+	Init() error
+
+	// GetName returns the name of the event handler
+	GetName() string
+
+	// GetNetworkPlugin ... logging driver says: "all"
+	GetNetworkPlugin() string
+
+	// Stop is called once during stop to let the network plugin cleanup anything necessary during
+	// at stop if at all. If the uninstall flag is true, the handler must cleanup everything because
+	// Submariner is being uninstalled from the system.
+	Stop(uninstall bool) error
+
+	// TransitionToNonGateway is called when there is a transition from GatewayNode to NonGateway on the
+	// This will be called only when the transition happens and not periodically.
```

Comment:
Sorry I'm not clear on this, specifically the phrase "in each node".  What exactly triggers this event? Is it for the node on which the handler's process is running? Or is it for any node that transitions, in which case shouldn't the method pass identifying node information?

Reply:
I rephrased your suggestion a bit, because what we want to say is... during than transition on the exact node the handler is running..., not on other nodes.

Reply:
OK then I think it should be phrased as "TransitionToNonGateway is called once for each transition of the local node from Gateway to a non-Gateway.".

Reply:
Done